### PR TITLE
Add Cso model and Orchid admin for Cso CRUD

### LINKED ISSUE #15: Create `Cso` model
This entity would contain information about each CSO in the directory.

Properties include:
- `name`
- `location`
- `assessment_score`
- `status` ( enum -> approved, pending, rejected)
- `partnership`
- `image`

### PR DESCRIPTION
**Pull request by ExcelDurant**
This Pull Request adds the `Cso` model and corresponding database migration. In addition, Orchid admin has been implemented to manage CRUD operations for `Cso` models. This includes the ability to create, edit, and delete `Csos`.

closes #15

### CODE REVIEW

File: database/migrations/2023_06_21_102051_create_cso_organization_leaderships_table.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('cso_organization_leaderships', function (Blueprint $table) {
+            $table->id();
+            $table->string('name')->default('Women headed organization');
```
Comment (Fenn-CS) [inline database/migrations/2023_06_21_102051_create_cso_organization_leaderships_table.php:16]: What does this mean?

File: database/migrations/2023_06_21_101536_create_cso_activity_domains_table.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('cso_activity_domains', function (Blueprint $table) {
+            $table->id();
+            $table->string('domain')->default('Women empowerment');
```
Comment (Fenn-CS) [inline database/migrations/2023_06_21_101536_create_cso_activity_domains_table.php:16]: Why would you put such defaults? Just make the `domain` required.

File: database/migrations/2023_06_20_092548_create_csos_table.php [hunk 3]
```diff
@@ -14,11 +14,34 @@ public function up(): void
         Schema::create('csos', function (Blueprint $table) {
             $table->id();
             $table->string('name');
-            $table->string('location');
             $table->integer('assessment_score');
             $table->enum('status', ['approved', 'pending', 'rejected'])->default('pending');
             $table->string('partnership');
             $table->string('image');
+            $table->string('acronym');
```
Comment (Fenn-CS) [inline database/migrations/2023_06_20_092548_create_csos_table.php:21]: All of these fields except `name` should be `nullable`

File: app/Models/CsoActivityDomain.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (Fenn-CS) [inline app/Models/CsoActivityDomain.php:1]: Why is this necessary? The activity domain can just be a field right?

File: app/Models/CsoOrganizationLeadership.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (Fenn-CS) [inline app/Models/CsoOrganizationLeadership.php:1]: Also why do you need a model for this? Could it just be a field as well?

**Review by Fenn-CS — CHANGES_REQUESTED** [review]
We don't want to create so many models especially if we don't know what information we would be tracking on them.

File: app/Models/CsoActivityDomain.php (hunk 4 above)
Comment (ExcelDurant) [reply app/Models/CsoActivityDomain.php:1]: The field for `domain of activity` are checkboxes. I thought it meant that multiple could be selected

Comment (ExcelDurant) [reply app/Models/CsoActivityDomain.php:1]: If only one `domain of activity` should be selected, it means maybe it has to be `radio` buttons instead of `checkbox`

Comment (Fenn-CS) [reply app/Models/CsoActivityDomain.php:1]: Okay

**Review by Fenn-CS — CHANGES_REQUESTED** [review]
Sorry about the `CsoActivity` initial confusion. I did reply you with okay when you explained why the model was there.


Last thing, look if the front-end pages for CSO listings are already available, if so you can just implement the cso listing as part of this PR

**Comment by ExcelDurant** [comment]
> Sorry about the `CsoActivity` initial confusion. I did reply you with okay when you explained why the model was there.
> 
> Last thing, look if the front-end pages for CSO listings are already available, if so you can just implement the cso listing as part of this PR

Okay understood

File: database/migrations/2023_06_20_092548_create_csos_table.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('csos', function (Blueprint $table) {
```
Comment (Fenn-CS) [inline database/migrations/2023_06_20_092548_create_csos_table.php:14]: Everything except name should be nullable

Comment (Fenn-CS) [reply database/migrations/2023_06_20_092548_create_csos_table.php:14]: name and timestamps of course

**Review by Fenn-CS — APPROVED** [review]
